dandischema/models.py

```diff
@@ -1088,16 +905,18 @@ class BareAssetMeta(CommonModel):
     encodingFormat: Union[HttpUrl, str] = Field(
         title="File encoding format", nskey="schema"
     )
-    digest: Dict[DigestType, str] = Field(default_factory=dict)
+    digest: Dict[DigestType, str] = Field(
+        title="A map of dandi digests to their values", nskey="dandi"
+    )
+    path: str = Field(nskey="dandi")
+
     dateModified: Optional[datetime] = Field(
         nskey="schema", title="Asset (file or metadata) modification date and time"
     )
     blobDateModified: Optional[datetime] = Field(
         nskey="dandi", title="Asset file modification date and time"
     )
 
-    path: str = Field(None, nskey="dandi")
-
```

Comment:
is order of specification here reflects itself in metadata records dumps?  then we might as well harmonize as to place all mandatory first (may be sort them alphabetically) and then all optional next, also alphabetically.

Special cases -- read-only identifier first?  the rest of read-only fields last.

Might be feasible to do automagically in that metaclass I proposed above. if done prior feeding the list of attributes to super class (so then `schemaKey` would be last, as it should ;)?)

Reply:
i wouldn't rely on ordering here i just moved it to be visually closer to mandatory fields